File: 2022/20221226/LCM/src/main/java/LCM.java
public class LCM {
    public int solution(int[] arr) {
        int answer = arr[0];

        for (int number : arr) {
            answer = calcLCM(answer, number);
        }
        return answer;
    }

    public int calcLCM(int a, int b) {
        int greatestCommonDivisor = calcGCD(a, b);
        return a * b / greatestCommonDivisor;
    }

    public int calcGCD(int a, int b) {
        while (b > 0) {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
}

File: 2022/20221227/LCM.js
function GCD(a, b) {
  return b ? GCD(b, a % b) : a;
}

function LCM(a, b) {
  return a * b / GCD(a, b);
}

function solution(arr) {
  let answer = arr[0];

  arr.forEach((number) => {
    answer = LCM(answer, number);
  });
  return answer;
}
